pkg/forwarder/forwarder_health.go

```diff
@@ -114,8 +115,21 @@ func (fh *forwarderHealth) setAPIKeyStatus(apiKey string, domain string, status
 	apiKeyStatus.Set(obfuscatedKey, status)
 }
 
+func getAPIValidationEndpoint(domain string) string {
+	apiDomain := ""
+	if strings.HasSuffix(domain, ".com") {
```

Comment:
Domain could be a custom URL set by the user (ie with a explicit proxy). Also we already have a logic to deduce the domain from the site: https://github.com/DataDog/datadog-agent/blob/master/pkg/config/config.go#L862. We should avoid having if/else for each DD zone since we're likely to add more of them.

Those URLs come from `config.GetMultipleEndpoints()` where we add the `https://api.` prefix. We might want to improve there in some way to have both the `api` and `app` prefix when using the `site` configuration.